dm/dm/master/openapi.go

```diff
@@ -929,6 +929,16 @@ func getOpenAPISubtaskStatusByTaskName(taskName string,
 			return nil, terror.ErrOpenAPICommonError.New("worker's query-status response is nil")
 		}
 		sourceStatus := workerStatus.SourceStatus
+		openapiSubTaskStatus := openapi.SubTaskStatus{
+			Name:       taskName,
+			SourceName: sourceStatus.GetSource(),
+			WorkerName: sourceStatus.GetWorker(),
+		}
+		if !workerStatus.Result {
```

Comment:
The logic here put construct `openapiSubTaskStatus` advance, and will lose other attributes like `Stage`, `Unit`. so just a question, does it meet expectations?

Reply:
oh, i see. `workerStatus.Result` is false when rpc to worker is fail. you test at integration test.